models/fact.js

```diff
@@ -2,40 +2,29 @@ var _ = require("underscore");
 var data = require("./data.js");
 var utils = require("../public/js/shared_utils.js");
 
-// http://stackoverflow.com/questions/2532218/pick-random-property-from-a-javascript-object
-function randomProperty(obj, pre) {
-  var result;
-  var count = 0;
-  for (var prop in obj) {
-    if (obj.hasOwnProperty(prop)) {
-      if (Math.random() < 1 / ++count) {
-        result = prop;
-      }
-    }
-  }
-  return result;
-}
-
-function getRandomApiNum(type, options) {
+function getRandomApiNum(options) {
   var min = parseInt(options.min, 10);
   var max = parseInt(options.max, 10);
+
   if (isNaN(min) && isNaN(max)) {
-    return utils.randomChoice(dataKeys[type]);
+    return utils.randomChoice(dataKeys[options.type]);
   } else {
     if (isNaN(min)) {
       min = -Infinity;
     } else if (isNaN(max)) {
       max = Infinity;
     }
+
     // TODO: Use binary search here instead of O(n) linear search
-    var valid_keys = _.filter(dataKeys[type], function (element) {
+    var valid_keys = _.filter(dataKeys[options.type], function (element) {
       return element >= min && element <= max;
     });
+
     return utils.randomChoice(valid_keys);
   }
 }
 
-function getSentence(wantFragment, number, type, data) {
+function getSentence({ wantFragment, number, type, data }) {
```

Comment:
yeah with this function signature `wantFragment` will be undefined if it's not provided, so you should never need to explicitly pass `wantFragment: undefined` in the tests.